assets/js/modules/analytics/util/index.js

```diff
@@ -80,11 +80,10 @@ function reduceAnalyticsRowsData( rows, selectedStats ) {
 		if ( row.metrics ) {
 			const { values } = row.metrics[ 0 ];
 			const dateString = row.dimensions[ 0 ];
-			const dateWithDashes =
-				dateString.slice( 0, 4 ) + '-' +
-				dateString.slice( 4, 6 ) + '-' +
-				dateString.slice( 6, 8 );
-			const date = new Date( dateWithDashes );
+			const year = dateString.slice( 0, 4 );
+			const month = Number( dateString.slice( 4, 6 ) ) - 1;
+			const day = dateString.slice( 6, 8 );
+			const date = new Date( year, month.toString(), day );
```

Comment:
Suggested clarification
```suggestion
			const monthIndex = Number( dateString.slice( 4, 6 ) ) - 1;
			const day = dateString.slice( 6, 8 );
			const date = new Date( year, monthIndex.toString(), day );
```
or
```suggestion
			const month = Number( dateString.slice( 4, 6 ) );
			const day = dateString.slice( 6, 8 );
			const date = new Date( year, ( month - 1 ).toString(), day );
```